src/main/java/com/hotels/beeju/extensions/BeejuJUnitExtension.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import java.io.File;
+import java.util.Map;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.thrift.TException;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import com.hotels.beeju.core.BeejuCore;
+
+abstract public class BeejuJUnitExtension implements BeforeEachCallback, AfterEachCallback {
+
+  public BeejuCore core;
+
+  public BeejuJUnitExtension(String databaseName, Map<String, String> configuration) {
+    core = new BeejuCore(databaseName, configuration);
+  }
+
+  /**
+   * This method can be overridden to provide additional initialisations.
+   * </p>
+   */
+  protected void init() throws Exception {
```

Comment:
Do we need `init()` if we have a `beforeEach()` which could be overridden? Do we use this feature ourselves anywhere? If not we probably don't need to make it available.